src/quicklist.c

```diff
@@ -154,7 +157,19 @@ void quicklistSetFill(quicklist *quicklist, int fill) {
     } else if (fill < -5) {
         fill = -5;
     }
-    quicklist->fill = fill;
+
+    if (fill >= 0) {
+        quicklist->limit_type = QUICKLIST_COUNT_LIMIT_TYPE;
+        quicklist->limit = (fill == 0) ? 1 : fill;
+    } else {
+        quicklist->limit_type = QUICKLIST_NODE_SIZE_LIMIT_TYPE;
+        quicklist->limit = quicklistNodeNegFillLimit(fill);
+    }
+}
+
+REDIS_STATIC void quicklistSetLimit(quicklist *quicklist, unsigned int limit_type, size_t limit) {
```

Comment:
```suggestion
REDIS_STATIC void quicklistSetLimit(quicklist *ql, unsigned int limit_type, size_t limit) {
```
it can be simpler to use ql, same as other places.